### PR TITLE
Cbnt prov/new generation

### PR DESCRIPTION
**Pull request by ChriMarMe**
Add support for SPR-SP CBnT implementation.

### CODE REVIEW

File: cmd/cbnt-prov/cmd.go [hunk 1]
```diff
@@ -167,13 +167,15 @@ type signKMCmd struct {
 	KmIn     string `arg required name:"kmin" help:"Path to the generated Key Manifest binary file." type:"path"`
 	KmOut    string `arg required name:"kmout" help:"Path to write the signed KM to"`
 	Key      string `arg required name:"km-keyfile" help:"Path to the encrypted PKCS8 private key file." type:"path"`
+	SignAlgo string `arg required name:"signalgo" help:"Signing algorithm for KM. E.g.: RSAASS, RSAPSS, SM2"`
```
Comment (walterchris) [inline cmd/cbnt-prov/cmd.go:170]: RSASSA

File: cmd/cbnt-prov/cmd.go [hunk 2]
```diff
@@ -927,19 +944,59 @@ func (p printFITCmd) Run(ctx *context) error {
...
+		return err
+	}
+
+	var km key.Manifest
+	r := bytes.NewReader(kmRaw)
+	_, err = km.ReadFrom(r)
+	if err != nil {
+		return err
+	}
+	if err := km.KeyAndSignature.Verify(kmRaw[:km.KeyAndSignatureOffset()]); err != nil {
```
Comment (walterchris) [inline cmd/cbnt-prov/cmd.go:959]: Why are we doing the "one-liner" here and above explicitly error checking?

File: cmd/cbnt-prov/cmd.go [hunk 3]
```diff
@@ -927,19 +944,59 @@ func (p printFITCmd) Run(ctx *context) error {
...
+		return err
+	}
+
+	var bpm bootpolicy.Manifest
+	r := bytes.NewReader(bpmraw)
+	_, err = bpm.ReadFrom(r)
+	if err != nil {
+		return err
+	}
+	if err := bpm.PMSE.Verify(bpmraw[:bpm.KeySignatureOffset]); err != nil {
```
Comment (walterchris) [inline cmd/cbnt-prov/cmd.go:978]: Same here.

**Review by walterchris — CHANGES_REQUESTED** [review]

File: cmd/cbnt-prov/cmd.go (hunk 1 above)
Comment (ChriMarMe) [reply cmd/cbnt-prov/cmd.go:170]: whoopsie

File: cmd/cbnt-prov/cmd.go (hunk 2 above)
Comment (ChriMarMe) [reply cmd/cbnt-prov/cmd.go:958]: done

**Comment by walterchris** [comment]
Verified to work on Archer City CRB. ACM establishes SRTM.

**Review by walterchris — APPROVED** [review]

**Review by ArthurHeymans — APPROVED** [review]

**Pull request merged**